components/table/TimesheetTable.tsx

```diff
@@ -18,108 +17,66 @@ import {
 import { Table, Tbody, Td, Th, Thead, Tr } from '@chakra-ui/table';
 import React, { useState } from 'react';
 import { useSWRConfig } from 'swr';
-import { FormStatus } from '../projects/NewProject/reducer';
 import * as fetch from '@/lib/utils/fetch';
 import ErrorAlert from '../common/ErrorAlert';
 import useData from '@/lib/hooks/useData';
 import Loading from '../common/Loading';
+import { FormStatus } from '../form/ProjectForm';
+import { employeeEndpointURL, timesheetEndpointURL } from '@/lib/const';
+import { TimesheetDTO, ProjectDTO, EmployeeDTO } from '@/lib/types/dto';
 
 type TimesheetTableProps = {
-    timesheets?: components['schemas']['TimesheetDTO'][];
```

Comment:
There is probably not a good reason for this to be nullable. Preferring conditional render
```
p && <Component p={p}/>
```
over a nullable `p` will almost certainly prevent quite a few future bugs